### PR TITLE
feat:  root manager new variables

### PR DESCRIPTION
**Pull request by gotzenx**
The storage variables needed for the Root Manager refactor.

CLOSES CON3-3

### CODE REVIEW

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 1]
```diff
@@ -223,6 +223,21 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
...
+   * @notice The list of valid aggregate roots for a given timestamp. Each time a new aggregate root is generated or
+   * finalized, it will be added to this mapping using the block.timestamp as key.
+   * @dev This is only used as Data-Availability for off-chain agents. Especially for the Watchers that will need to fetch
+   * from this contract the correct aggregate root to verity that the data proposed on the Spoke Connectors is correct.
+   * @dev rooTimestamp => aggregateRoot
+   */
+  mapping(uint256 => bytes32) public validAggregateRoots;
+
+  /**
+   * @notice Timestmap of the last saved aggregate root.
```
Comment (0xng) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:236]:
```suggestion
   * @notice Timestamp of the last aggregate root saved.
```

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 2]
```diff
@@ -223,6 +223,21 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
...
+   * @dev This is mainly used on to ensure that the propagate function will always send the latest aggregate root available.
```
Comment (0xng) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:237]:
```suggestion
   * @dev Used to ensure the propagate function will send the latest aggregate root available.
```

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 3]
```diff
@@ -223,6 +223,21 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
    */
   mapping(address => bool) public allowlistedProposers;
 
+  /**
+   * @notice The list of valid aggregate roots for a given timestamp. Each time a new aggregate root is generated or
```
Comment (0xng) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:227]: Everything after the period/first sentence should be a @dev tag imo.

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 4]
```diff
@@ -223,6 +223,21 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
...
+   * finalized, it will be added to this mapping using the block.timestamp as key.
+   * @dev This is only used as Data-Availability for off-chain agents. Especially for the Watchers that will need to fetch
```
Comment (0xng) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:229]:
```suggestion
   * @dev used as Data-Availability for off-chain agents. Especially for the Watchers that fetch the correct aggregate root from this contract in order to verify the data proposed on the Spoke Connectors.
```

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 5]
```diff
@@ -223,6 +223,21 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
...
+   * from this contract the correct aggregate root to verity that the data proposed on the Spoke Connectors is correct.
+   * @dev rooTimestamp => aggregateRoot
```
Comment (0xng) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:231]:
```suggestion
   * @dev rootTimestamp => aggregateRoot
```

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 6]
```diff
@@ -223,6 +223,21 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
...
+   */
+  mapping(uint256 => bytes32) public validAggregateRoots;
+
+  /**
+   * @notice Timestmap of the last saved aggregate root.
+   * @dev This is mainly used on to ensure that the propagate function will always send the latest aggregate root available.
+   */
+  uint256 lastSavedAggregateRootTimestamp;
```
Comment (gotzenx) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:239]: it should be public

**Review by 0xng — APPROVED** [review]

**Pull request merged**